### ISSUE TITLE
cast string to int

### ISSUE THREAD

**Issue by knsong** [NVIDIA/DALI#1503]
hi,
Is there a way to cast label encoded as a string  in a `tfrecord` file back to int type in DALI pipeline?

**Comment by JanuszL** [comment]
Hi,
Now DALI reads a string and creates a `uint8_t` tensor from that. Do I understand you correctly you want to have a single value created from it?

**Comment by knsong** [comment]
> Hi,
> Now DALI reads a string and creates a `uint8_t` tensor from that. Do I understand you correctly you want to have a single value created from it?

Thanks for the answer and yes, I want read the string as a serialized number by tfrecord. e.g. the label is encoded as a string when I created the tfrecords:
`features={'label': tf.FixedLenFeature([], tf.string), 
                   'data': tf.FixedLenFeature([], tf.string) })`
and now I want to decode the `label` which should be int type from the corresponding string in DALI tfrecord pipeline.

**Comment by JanuszL** [comment]
@knsong - have you tried `as_string` from tensorflow?

**Comment by knsong** [comment]
> @knsong - have you tried `as_string` from TensorFlow?

No, I'm using pytorch and DALI tfrecord pipeline.

**Comment by JanuszL** [comment]
We haven't explored this, so what I would check myself is:
- how applicable is numpy tostring method
- try to develop own uint8 to string converter

**Comment by knsong** [comment]
> We haven't explored this, so what I would check myself is:
> 
> * how applicable is numpy tostring method
> * try to develop own uint8 to string converter

Ok, I finaly change the tfrecord converter pipelins. Thanks
